core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -158,6 +158,37 @@ private[spark] class HostLocalDirManager(
   }
 }
 
+private[spark] trait MigratableRDDResolver {
```

Comment:
Having a private trait that only ever has one implementation seems weird. I think that you are trying to reduce the API scope that's accessible to the `BlockManagerDecommissionManager`, but I don't think this beneficial.

Reply:
Well I think it helps a lot in understanding the code. It states cleanly what to expect from the other object. It is not only isolating but it channels further API calls into this interface.